libvips/foreign/jxlload.c

```diff
@@ -492,8 +486,15 @@ vips_foreign_load_jxl_read_frame(VipsForeignLoadJxl *jxl, VipsImage *frame,
 	size_t buffer_size;
 	JxlDecoderStatus status;
 
-	if (jxl->frame_no >= frame_no)
-		return 0;
+	int skip = frame_no - jxl->frame_no - 1;
+	if (skip > 0) {
+#ifdef DEBUG_VERBOSE
+		printf("vips_foreign_load_jxl_read_frame: skipping %d frames\n",
+			skip);
+#endif /*DEBUG_VERBOSE*/
+		JxlDecoderSkipFrames(jxl->decoder, skip);
+		jxl->frame_no = frame_no - 1;
```

Comment:
Or maybe `jxl->frame_no += skip`? It's a little clearer.

Reply:
Indeed, that's clearer. Incorporated with commit 4c8dde602d14171c57d466ea97c24f3348c7b166.